managed/ui/src/components/universes/UniverseDetail/UniverseDetail.js

```diff
@@ -186,12 +187,26 @@ class UniverseDetail extends Component {
       showGFlagsModal,
       showManageKeyModal,
       showDeleteUniverseModal,
+      showToggleUniverseStateModal,
       closeModal,
       customer,
       customer: { currentCustomer },
       params: { tab }
     } = this.props;
     const { showAlert, alertType, alertMessage } = this.state;
+    const clusters = universe?.currentUniverse?.data?.universeDetails?.clusters;
+    const universePaused = universe?.currentUniverse?.data?.universeDetails?.universePaused;
+
+    // This variable will store the universe provider type which helps to enable
+    // Pause Universe functionality.
+    // TODO: For now, we're enabling the Pause Universe for providerType==='aws'
+    // only. This functionality needs to be enabled for all the cloud providers and
+    // once that's done this needs to be removed.
+    const providerType = isNonEmptyArray(clusters)
+      ? clusters.map((cluster) => {
+        return cluster?.userIntent?.providerType;
+      })
+      : null;
```

Comment:
having a hard time getting this logic
if you need to know if the primary cluster has `aws` provider type - consider smth like this
https://github.com/yugabyte/yugabyte-db/blob/6ade350a826d9a97ce6191a6b081bc23c2f75c3e/managed/ui/src/utils/UniverseUtils.js#L139-L142

Reply:
Done.